### DISCUSSION TITLE
stucked fake session connections.

### DISCUSSION THREAD

**Discussion by gerikov** [WalletConnect/walletconnect-monorepo#3872] [Developer Support]
Dear WalletConnect team,

we are developing our dapps and wallet and facing a little bit wierd issue and it happens only a few times.

It seems from the client side the connection is fine, but we cannot execute any transaction from there. The problem comes up if i try to connect, disconnect, reconnect multiple times in short term. It seems like the session is cached at the client side but its just a fake connection and all tx will fail, the popups never apper to approve it.

The isDisconnected flag is false, the address is also availble from the useAccount hook.

Any idea how can I detect these fake connections, is it possible to make a query to the server with the sessionID to be sure is it active or not?

Try to find a solution for 2 days, but im out of the ideas. Unfortunately I cannot share the dapp neither wallet, we will make the application next week.

Thanks,
G

**Comment by glitch-txs** [comment]
Can you check that you're not initiating the clients twice? Both in the dapp and the wallet.
If the issue still persists, please share a MRE that we can reproduce.

**Reply by maryannabt** [reply]
@gerikov @glitch-txs  Hello, did you find a solution to this? If yes, please share, I'm experiencing exactly the same thing. Thank you! (Trying to connect/disconnect and make transactions in Safe Wallet, using web3modal wagmi).
